architecture/namespaced-attributes.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Namespaced Attributes
+nav_order: 4
+parent: Architecture
+---
+
+# Namespaced Attributes
+
+Good
+{: .label .label-green}
+
+```html
+<!-- index.html -->
+<input data-controller="filter" data-filter-param-category="cats" data-filter-param-rating="5" data-filter-param-color="black" type="text" data-action="input->filter#update">
+```
+{: .border-green}
+
+```js
+// filter_controller.js
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  update () {
+    const url = new URL(window.location)
+    Object.keys(Object.assign({}, this.element.dataset))
+      .filter(attr => attr.startsWith("filterParam"))
+      .forEach(attr => {
+        url.searchParams.set(
+          attr.slice(11).replace(/^\w/, c => c.toLowerCase()),

```

Comment:
The problem is that it really is arbitrary! My goal here was to demonstrate a sort of "second-order" namespacing, which is governed by no convention. I could suggest a convention, but it might well be entirely different depending on whether you're building something for people to download from npm or a component for a project.

Really am open to suggestions, here. More description is likely the easiest fix.